src/config.rs

```diff
@@ -272,6 +278,53 @@ impl Config {
     }
 }
 
+#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
+pub enum ClippyPreference {
+    /// Disable clippy
+    Off,
+    /// Enable clippy, but "allow" clippy lints (ie require "warn" override)
+    OptIn,
+    /// Enable clippy
+    On,
+}
+
+impl FromStr for ClippyPreference {
+    type Err = ();
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s.to_lowercase().as_str() {
+            "off" => Ok(ClippyPreference::Off),
+            "optin" | "opt-in" => Ok(ClippyPreference::OptIn),
+            "on" => Ok(ClippyPreference::On),
+            _ => Err(()),
+        }
+    }
+}
+
+/// Case-insensitive deserialization for `ClippyPreference`
+fn deserialize_clippy_preference<'de, T, D>(deserializer: D) -> Result<T, D::Error>
```

Comment:
Could you use the `rename` serde attribute instead of writing a custom deserialiser?

Reply:
I was going for permissive deserialization, not wanting it to fail because someone had typed "Optin". By the looks of it serde's metaprograming seems only built for strictness.

Reply:
Would you prefer to just standardize on, say, kebab-case?

Reply:
That makes sense. I think going for permissive is fine here. Could you add a comment explaining that you're doing that rather than doing regular Serde stuff please?

Reply:
I've clarified it more now, plus added a unit test for `FromStr` logic